frontend/README.md

```diff
@@ -0,0 +1,389 @@
+        Image Provider->>+Image Gateway: Fetch images from API (paginated)
+        Image Gateway->>+Models: Generate image models
+        Models-->>-Image Gateway: Return Image Models
+        Image Gateway-->>-Image Provider: Returns Images
+        Image Provider-->>-Main Page: Updates Image List
+        Main Page-->>-User: Renders Image list
+    end
+    User->>+Main Page: Clicks Upload
+    Main Page-->>-User: Shows Controls
+    User->>+Main Page: Selects Image
+    Main Page->>+Image Provider: Triggers Upload
+    Image Provider->>+Image Provider: Gets image metadata
+    Image Provider->>+Image Provider: Validates File Metadata
+    Image Provider->>+Models: Generates upload model
+    Models-->>-Image Provider: Upload Model
+    Image Provider->>+Image Provider: Updates Upload Map
+    par Image Provider to Main Page
+        Image Provider-->>-Main Page: Updates in progress uploads
+        Main Page-->>-User: Shows In progress upload
+    and Image Provider to Image Gateway
+        Image Provider->>+Image Gateway: Get Singed Url
+        Image Gateway->>+Models: Generates response model
+        Models-->>-Image Gateway: Response model
+        Image Gateway-->>-Image Provider: Signed URL
+        Image Provider->>+Image Gateway: Starts Upload
+        loop on progress
+            Image Gateway->>+Image Provider: Update Progress
+            Image Provider-->>+Main Page: Updates in progress uploads
+            Main Page-->>-User: Updates upload progress
+        end
+        Image Gateway-->-Image Provider: Finish Upload
+        par Image Provider to Image Provider
+            Image Provider->>+Image Provider: Updates Upload Map
+            Image Provider->>+Main Page: Shows Notification
+            Image Provider-->>+Main Page: Updates in progress uploads
+            Main Page-->>-User: Removes in progress upload
+        and Image Provider to Image Gateway
+            Image Provider->>+Image Gateway: Refresh List
+            Image Gateway->>+Models: Generate image models
+            Models-->>-Image Gateway: Image Models list
+            Image Gateway-->>-Image Provider: Return Images
+            Image Provider-->>+Main Page: Updates Image List
+            Main Page-->>-User: Renders Image list
+        end
+    end
+```
+
+### Edge Cases
+
+- What if the user selects an invalid asset type, e.g a video instead of an image? When displaying the native controls, the input should filter the assets to just accept the valid mimetypes, e.g image/jpg, image/png, etc. Aside from that, the file object should be validated before fetching the signed url.
+- What happens in case the upload fails? App should display the upload status as failed and allow users to retry.
+- What happens if the user moves to another side half-way during the upload? Navigation should be blocked that way if the user wants to move away there should be an alert to confirm the action, in case the user moves away the upload is lost.
```

Comment:
```suggestion
- What happens if the user moves to another site half-way during the upload? Navigation should be blocked that way if the user wants to move away there should be an alert to confirm the action, in case the user moves away the upload is lost.
```
This one is under debate for us because the back-end would still receive the full upload and resolve, we believe.

Reply:
So that's a requirement, having the upload continue if the user moves away from the app? I believe that feels really strange as someone looking at an ephemeral web application, you can consider things like that for electron apps or even mobile apps.

I would ask you to look at any social media platform that does that, it can be even a possible privacy issue if you upload things in the background if the user moves away.